advanced/server-extension/setupbase.py

```diff
@@ -0,0 +1,681 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+"""
+This file originates from the 'jupyter-packaging' package, and
+contains a set of useful utilities for including npm packages
+within a Python package.
+"""
```

Comment:
Unfortunately, `setup.py` file is specific in the sense that any package imported in it should be present before _using_ the package source.

This is the reason the `jupyter-packaging` README states:

>The contents of jupyter_packaging.py are copied locally to setupbase.py so they are available to run the setup.py script itself.

So we have two options:

Option | Pros | Cons
--- | --- | ---
Direct copy | usable directly - no need to install an easy-to-forget setup requirement | Don't benefit of updates (I don't think they will be frequent)
As setup requirement | Up-to-date with a reference | May break the API or the available methods

Personally, I prefer to keep it as a direct copy.

@jtpio what do you think?

Reply:
Usually `setupbase.py` is indeed copied between projects (or from `jupyter-packaging`).

So we can do the same here, and maybe add a small paragraph in the example README (in the packaging section) to explain why we need this file. Since it's a lot of code it could be confusing to newcomers.

Reply:
Thx for info. Let's keep the copy with a bit more explanation in the README.

Reply:
I push a note in https://github.com/jtpio/jupyterlab-extension-examples/pull/46/commits/7bcbee6008a2a94ad5df1e1a4410054ddbd76a23

@echarles @jtpio What do you think?

Reply:
Looks good :+1:

Reply:
:+1: